src/Courses/CourseDetailsPage.tsx

```diff
@@ -34,7 +35,9 @@ export const CourseDetailsPage: React.FC<CourseDetailsPageProps> = () => {
     const { courseId } = useParams();
     const [course, setCourse] = useState<CourseObject>(new CourseObject());
     const [activeTab, setActiveTab] = useState<CourseDetailsTabs>(CourseDetailsTabs.DETAILS);
-    const userType: UserRole = getUserRole(Cookies.get(CookieEnum.USERTYPE));
+    const userType: UserRole = getUserRole();
+    const userId: number = getUserId();
+    const [studentNameAndId, setStudentNameAndId] = useState<{name: string, userId: number} | null>(null);
```

Comment:
I don't like that we have two attributes in the same hook, if one were to change it wouldn't affect the other, granted taht shouldn't happen but I don't know how much of an advantage this really gives you

Reply:
Follow up: would it better to use a 'me' parameter on the backend?